subsys/bluetooth/mesh/statistic.h

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_SUBSYS_BLUETOOTH_MESH_STATISTIC_H_
+#define ZEPHYR_SUBSYS_BLUETOOTH_MESH_STATISTIC_H_
+
+void bt_mesh_stat_planned_count(struct bt_mesh_adv *adv);
+void bt_mesh_stat_succeeded_count(struct bt_mesh_adv *adv);
+void bt_mesh_stat_rx(enum bt_mesh_net_if net_if);
```

Comment:
Related to [this comment](https://github.com/zephyrproject-rtos/zephyr/pull/59962#pullrequestreview-1511372053), you could make these empty based on the `kconfig` option. This would eliminate the need to check the option in all the calls to these functions.

```c
#ifdef CONFIG_BT_MESH_STATISTIC

void bt_mesh_stat_planned_count(struct bt_mesh_adv *adv);
void bt_mesh_stat_succeeded_count(struct bt_mesh_adv *adv);
void bt_mesh_stat_rx(enum bt_mesh_net_if net_if);

#else

#define bt_mesh_stat_planned_count(...)   ((void)0)
#define bt_mesh_stat_succeeded_count(...) ((void)0)
#define bt_mesh_stat_rx(...)              ((void)0)

#endif
```

Reply:
It breaks self-explanatory of source code. You never know at the moment this is function or macro. I strictly disagree with this practice.